crypto/fipsmodule/service_indicator/service_indicator.c

```diff
@@ -57,7 +65,8 @@ uint64_t FIPS_service_indicator_after_call(void) {
 }
 
 int FIPS_service_indicator_check_approved(uint64_t before, uint64_t after) {
-  if(before != after) {
+
+  if (before < after) {
```

Comment:
I remember that we did `(before != after)` to avoid `after` overflowing and becoming less than `before`

Reply:
can we then remove the check in line 88, in update_state function? If the check is `before != after` why do we have a counter then, instead of a `{0, 1}` flag?

Reply:
Yeah, now that you point that out, line 88 was a recent change when we were trying to fix underflow/overflow errors with the service indicator locks. After our internal discussion, I think it's alright to do `before < after` if we're checking for overflowing there.

Reply:
I'm now worried about a service that has used more than 2^64 FIPS operations since its start, wouldn't the check on l.87 fail when the counter = 2^{64} - 1? I think that's why we didn't have that check and we were only checking if the value changed. (Having that check on the lock makes sense because our call graph is few levels deep.)

Reply:
yes, our current design allows only 2^64 FIPS approved operations, after that, it would always indicate "not approved". That's why my suggestion is to remove the check in line 87 and keep the `before != after` condition. This way we don't have the artificial limit on the number of FIPS approved operations and we don't have to care about overflowing the counter.

p.s. the counters are thread local so assuming that one FIPS approved operation takes 1 ns, it will take almost 600 years to overflow the counter :)

Reply:
I agree with you, Dusan in keeping the condition as `!=` and removing the check in l.87.
Good observation and the calculation :)

Reply:
done